pylint/extensions/check_ungrouped_isinstances.py

```diff
@@ -0,0 +1,46 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""Module to check ungrouped isinstances methods for pylint. """
+
+import astroid
+
+from pylint import checkers, utils
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+
+MSGS = {
+    'E2685': ("isinstances of '%s' are not grouped. "
+              "Use 'isinstance(%s, (class1, class2, ...))' instead",
+              "ungrouped-isinstances",
+              "Used when isinstances are not grouped by object"),
+}
+
+
+class UngroupedIsinstancesChecker(checkers.BaseChecker):
+    __implements__ = IAstroidChecker
+    name = 'ungrouped_isinstances'
+    msgs = MSGS
+
+    @check_messages(*(MSGS.keys()))
+    def visit_boolop(self, node):
+        "Check ungrouped isinstances"
+        if node.op != 'or':
+            return
+        first_args = [
+            value.args[0].as_string()
+            for value in node.values
+            if isinstance(value, astroid.Call) and value.args and
+            value.func.name == 'isinstance']
+        for duplicated_name in utils.get_duplicated(first_args):
```

Comment:
You have to check it is also a builtin. Also, let the inference do its job here, infer what value.func is and then check the name, only if it is a builtin.